db/dcrpg/pgblockchain.go

```diff
@@ -32,13 +33,14 @@ var (
 // ChainDB provides an interface for storing and manipulating extracted
 // blockchain data in a PostgreSQL database.
 type ChainDB struct {
-	db                 *sql.DB
-	chainParams        *chaincfg.Params
-	devAddress         string
-	dupChecks          bool
-	bestBlock          int64
-	lastBlock          map[chainhash.Hash]uint64
-	addressCounts      *addressCounter
+	db            *sql.DB
+	chainParams   *chaincfg.Params
+	devAddress    string
+	dupChecks     bool
+	bestBlock     int64
+	lastBlock     map[chainhash.Hash]uint64
+	addressCounts *addressCounter
+	client        *rpcclient.Client
```

Comment:
I do not like adding rpcclient.Client to `ChainDB`.  I'd prefer a new type that embeds `ChainDB` and adds the client.  Thus, all the new ChainDB methods you added would then belong to that new type instead.